src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/TaskExtensions.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CommonLanguageServerProtocol.Framework
+{
+    internal static partial class TaskExtensions
```

Comment:
is it not possible to link in this file?  i worry about the code duplication here.

Reply:
I had difficulties doing so. In the original version of this file, there is use of the Contract.ThrowIfNull helper in a couple places. Pulling in that dependency, ended up pulling in multiple other source files which I had originally in the other PR and caused other problems (specifically build errors around multiply defined NullabilityAttributes).

I should point out that this assembly might not be long-lived for the Roslyn repository and has intentionally taken minimal dependencies. NoThrowAwaitable seemed useful enough to warrant the code duplication, whereas all the Contract classes and the resulting dependencies did not.